routing/dht/lookup.go

```diff
@@ -40,9 +40,12 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key key.Key) (<-chan pe
 		peerset.Add(p)
 	}
 
+	// since the query doesnt actually pass our context down
+	// we have to hack this here. whyrusleeping isnt a huge fan of goprocess
```

Comment:
creating a process from a context, or a context from a process, is the same thing as deriving a new context from a parent context, or a process from a parent process. it looks a bit awkward because the models are a bit different, but it is the same idea. it's a process tree. it's just that the process can let you reason about teardown, whereas the context cant.